### PR TITLE
distinguish Eval/Walk errors

### PR DESCRIPTION
**Pull request by tbg**

### CODE REVIEW

File: sql/parser/eval.go [hunk 1]
```diff
@@ -455,9 +455,12 @@ func EvalExpr(expr Expr, env Env) (Datum, error) {
 
 	case *CastExpr:
 		return evalCastExpr(t, env)
+
+	default:
+		return null, fmt.Errorf("eval: unsupported expression type: %T", expr)
 	}
 
-	return null, fmt.Errorf("unsupported expression type: %T", expr)
+	return null, fmt.Errorf("eval: unexpected expression: %T", expr)
```
Comment (vivekmenezes) [inline sql/parser/eval.go]: I don't see a point to the default above.

LGTM

Comment (tschottdorf) [inline sql/parser/eval.go]: the default means that we've forgotten to handle an expression in the switch, the other means that the expression was inspected and found to be unexpected.

Comment (vivekmenezes) [inline sql/parser/eval.go]: well, as the code stands the second return will never be called

Comment (tschottdorf) [inline sql/parser/eval.go]: it will. See for example `case *Subquery:` above.

Comment (tamird) [inline sql/parser/eval.go]: yeah the second return may as well be a panic to guard against a missing `return`?

Comment (tamird) [inline sql/parser/eval.go]: or the default should be a panic, like we do in `walk`?

Comment (tschottdorf) [inline sql/parser/eval.go]: yeah, doing a `panic` in default sounds reasonable. I'll do that.

**Comment by petermattis** [comment]
LGTM

**Pull request merged**